$NetBSD: patch-libmysql_CMakeLists.txt,v 1.4 2019/10/14 20:19:29 adam Exp $

Do not install libmysqlclient with the server.
Generate versioned lib symlinks on FreeBSD to supress PLIST divergence.
authentication_ldap and libmysql_api_test require libmysqld.

--- libmysql/CMakeLists.txt.orig	2019-12-09 19:53:17.000000000 +0000
+++ libmysql/CMakeLists.txt
@@ -241,13 +241,13 @@ IF(WIN32)
   ADD_SUBDIRECTORY(authentication_win)
   LIST(APPEND LIBS_TO_MERGE auth_win_client)
 ENDIF()
-
-# LDAP authentication SASL client plugin
-MESSAGE(STATUS "Creating LDAP authentication SASL client library.")
-ADD_SUBDIRECTORY(authentication_ldap)
   
 # Merge several convenience libraries into one big mysqlclient
+IF(WITHOUT_SERVER)
 MERGE_CONVENIENCE_LIBRARIES(mysqlclient ${LIBS_TO_MERGE} COMPONENT Development)
+ELSE()
+MERGE_CONVENIENCE_LIBRARIES(mysqlclient SKIP_INSTALL ${LIBS_TO_MERGE} COMPONENT Development)
+ENDIF()
 TARGET_LINK_LIBRARIES(mysqlclient PRIVATE ${LIBS_TO_LINK})
 
 # Visual Studio users need debug  static library for debug projects
@@ -283,12 +283,21 @@ ENDIF()
 
 # Merge several convenience libraries into one big mysqlclient
 # and link them together into shared library.
+IF(WITHOUT_SERVER)
 MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
   EXPORTS
   ${CLIENT_API_FUNCTIONS}
   ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
   ${CLIENT_API_NONBLOCKING_FUNCTIONS}
   COMPONENT SharedLibraries)
+ELSE()
+MERGE_LIBRARIES_SHARED(libmysql SKIP_INSTALL ${LIBS_TO_MERGE}
+  EXPORTS
+  ${CLIENT_API_FUNCTIONS}
+  ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
+  ${CLIENT_API_NONBLOCKING_FUNCTIONS}
+  COMPONENT SharedLibraries)
+ENDIF()
 TARGET_LINK_LIBRARIES(libmysql PRIVATE ${LIBS_TO_LINK})
 
 IF(WIN32)
@@ -302,7 +311,7 @@ ENDIF()
 
 IF(UNIX)
   # libtool compatability
-  IF(FREEBSD OR APPLE)
+  IF(APPLE)
     SET(OS_SHARED_LIB_VERSION "${SHARED_LIB_MAJOR_VERSION}")
   ELSE()
     SET(OS_SHARED_LIB_VERSION
@@ -366,6 +375,7 @@ CONFIGURE_FILE(api_test.c.in ${CMAKE_CUR
 # from @CLIENT_API_FUNCTIONS@ are declared by <mysql.h>. It will fail
 # to run if not all of these symbols are exported by the library.
 #
+IF(NOT WITHOUT_SERVER)
 MYSQL_ADD_EXECUTABLE(libmysql_api_test
   ${CMAKE_CURRENT_BINARY_DIR}/api_test.c
   LINK_LIBRARIES libmysql ${LIBRT}
@@ -386,3 +396,4 @@ ADD_CUSTOM_COMMAND(TARGET libmysql_api_t
   COMMAND libmysql_api_test
   > ${CMAKE_CURRENT_BINARY_DIR}/libmysql_api_test.out
   )
+ENDIF()
